webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/policies/NodeConfigData.scala

```diff
@@ -545,18 +550,22 @@ class TestNodeConfiguration(prefixTestResources: String = "") {
   val t6 = System.currentTimeMillis()
   NodeConfigData.logger.trace(s"Technique repository    : ${t6-t5} ms")
 
-  val draftServerManagement = new PolicyServerManagementService() {
-    override def setAuthorizedNetworks(policyServerId:NodeId, networks:Seq[String], modId: ModificationId, actor:EventActor) = ???
-    override def getAuthorizedNetworks(policyServerId:NodeId) : Box[Seq[String]] = Full(List("192.168.12.0/24", "192.168.49.0/24", "127.0.0.1/24"))
-    override def deleteRelaySystemObjectsPure(policyServerId: NodeId): IOResult[Unit] = ???
-    override def updateAuthorizedNetworks(policyServerId: NodeId, addNetworks: Seq[String], deleteNetwork: Seq[String], modId: ModificationId, actor: EventActor): Box[Seq[String]] = ???
+  val policyServerManagement = new PolicyServerManagementService() {
+    override def getPolicyServers(): IOResult[PolicyServers] = {
+      PolicyServers(
+        PolicyServer(Constants.ROOT_POLICY_SERVER_ID, List("192.168.12.0/24", "192.168.49.0/24", "127.0.0.1/24").map(s => AllowedNetwork(s, s"name for " + s)))
+        , Nil
+      ).succeed
+    }
+    override def updatePolicyServers(commands: List[PolicyServersUpdateCommand], modId: ModificationId, actor: EventActor): IOResult[PolicyServers] = ???
```

Comment:
I don't understand the purpose of `???`

Reply:
it's a placeholder that compiles but throws a "non implemented exception" if that code path is reached. It is usefull when you want to just implement some parts of a service for a test, but some methods are not needed in that test. 

It should never appears elsewhere than in test, and even in tests, it may mean that there is not a good separation of concerns between services, and that some services are too big and should be split over (but then, you are in usual balance research between atomicity and ease of use). So keeps eyes open when you see that, it may be the next refactoring.